rotary_utils/repair.py

```diff
@@ -55,59 +54,222 @@ def main(args):
+
+
+class RepairPaths:
+    """
+    A class containing file paths that are used during the assembly repair process.
+    """
+
+    def __init__(self, output_dir: str):
+        """
+        Instantiate a RepairPaths object.
+
+        :param output_dir: output directory for the assembly repair process.
+        """
+        """
+        Description of attributes:
+            linking_outdir_base: Temporary directory where re-assembly and stitching files will be saved.
+            circular_contig_tmp_fasta: Temporary FastA file containing only circular contigs to be stitched.
+            bam_filepath: path to a BAM file with mapping information of long reads to the contigs (it is OK if this
+                          file also contains mappings to other contigs outside those in the input file).
+            circlator_logs: path where a summary of log files from circlator will be saved.
+            end_repaired_contigs_filepath: path where the end-repaired contigs will be saved.
+            end_repair_report_filepath: path where the end repair report will be saved.
+            verbose_logfile: path to a logfile where shell script logs will be saved.
+        """
+        self.linking_outdir_base = os.path.join(output_dir, 'contigs')
+        self.circular_contig_tmp_fasta = os.path.join(self.linking_outdir_base, 'circular_input.fasta')
+        self.bam_filepath = os.path.join(output_dir, 'long_read.bam')
+        self.circlator_logs = os.path.join(output_dir, 'circlator_logs')
+        self.end_repaired_contigs_filepath = os.path.join(output_dir, 'repaired.fasta')
+        self.end_repair_report_filepath = os.path.join(output_dir, 'repaired_info.tsv')
+        self.verbose_logfile = os.path.join(output_dir, 'verbose.log')
+
+
+class StitchDirectories:
+    """
+    A class containing directory paths that are used during a single iteration of the assembly and stitch process.
+    """
+
+    def __init__(self, linking_outdir: str, length_threshold: int = None, make_dirs: bool = False):
+        """
+        Instantiate a StitchPaths object.
+
+        :param linking_outdir: output directory for the overall analysis.
+        :param length_threshold: length (bp) that the original contig will be subset to for the given analysis.
+        :param make_dirs: if True, makes the directories whose paths are specified by the object, if the directories
+                          do not already exist.
+        """
+        """
+        Description of constant attributes:
+            linking_outdir: output directory for the overall analysis.
+            log_dir_base: directory where key log files will be saved.
+            length_outdir: output directory for a specific length threshold.
+            log_dir: directory where log files for a specific length threshold will be saved.
+        """
+        self.linking_outdir = linking_outdir
+        self.log_dir_base = os.path.join(linking_outdir, 'logs')
+        if make_dirs:
```

Comment:
In this newly created `StitchDirectories` class, I add a flag so that the directories stored in the object can be created when an object is instantiated. This saves a couple lines of code in `iterate_linking_contig_ends` below, but is this best practice?

Reply:
Good to know -- thanks for confirming.